src/header.rs

```diff
@@ -0,0 +1,191 @@
+use tables::*;
+use Mp3Error;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Version {
+    Mpeg2_5,
+    Reserved,
+    Mpeg2,
+    Mpeg1,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Layer {
+    LayerI,
+    LayerII,
+    LayerIII,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Mode {
+    Stereo,
+    JointStereo,
+    DualChannel,
+    Mono,
+}
+
+/*
+|------|----------------|------------------------------|
+|      | Layer I and II |          Layer III           |
+|------|----------------|------------------|-----------|
+| bits |  Layer I & II  | Intensity Stereo | MS Stereo |
+|------|----------------|------------------|-----------|
+|  00  | bands 4 to 31  |        Off       |    Off    |
+|  01  | bands 8 to 31  |        On        |    Off    |
+|  10  | bands 12 to 31 |        Off       |    On     |
+|  11  | bands 16 to 31 |        On        |    On     |
+|------|----------------|------------------|-----------|
+*/
+#[derive(Debug, Clone, PartialEq)]
+pub enum ModeExtension {
+    Bands(u8),
+    Stereo(bool, bool),
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Emphasis {
+    None,
+    Ms50_15, // 50/15 ms
+    Reserved,
+    CcittJ17, // CCITT J.17
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct FrameHeader {
+    // MPEG Audio version
+    version: Version,
+    // MPEG layer
+    pub layer: Layer,
+    // Indicates that frame is protected by CRC (16 bit CRC follows header).
+    protection: bool,
+    // Bit rate
+    pub bitrate: u16,
+    // Sampling rate
+    pub sampling_rate: u16,
+    // Indicates that frame is padded with one extra slot (32 bits for Layer I, 8 bits for others).
+    pub padding: bool,
+    // Bit for application-specific triggers.
+    private: bool,
+    // Channel Mode
+    mode: Mode,
+    // Only used in Joint stereo mode to join informations that are of no use for stereo effect.
+    mode_extension: ModeExtension,
+    // Indicates that audio is copyrighted.
+    copyright: bool,
+    // Indicates that the frame is located on its original media.
+    original: bool,
+    // Tells the decoder that the file must be de-emphasized. Rarely used.
+    emphasis: Emphasis,
+}
+
+pub fn parse_frame_header(data: &[u8]) -> Result<FrameHeader, Mp3Error> {
+    let header = &data[..4];
+
+    // Sync word check
+    if (header[0] != 0xff_u8) && (header[1] < 0xe0_u8) {
```

Comment:
> could you replace the < with a & and != 

Why?

Reply:
using < for this task is kind of a hack imo, especially as it can only be used in a signed environment with masks of 111's at the highest bits and 0 otherwise.